components/FeatureHighlight/tests/unit/FeatureHighlightViewControllerTests.m

```diff
@@ -69,4 +87,40 @@ - (void)testTraitCollectionDidChangeBlockCalledWithExpectedParameters {
   XCTAssertEqual(passedTraitCollection, fakeTraitCollection);
 }
 
+- (void)testFeatureHighlightViewControllerDynamicTypeBehavior {
```

Comment:
Please add tests that assert the behavior of the intersection of the two APIs.  It's not clear to me how they would work together given their independent storage in the current version of this PR.

| adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable | mdc_legacyFontScaling | Font Scaling Type|
|---|---|---|
|N | N | N |
|N | Y | ?? |
|Y | N | ?? |
|Y | Y | Y |